sources/models/src/lib.rs

```diff
@@ -228,10 +228,14 @@ struct KubernetesSettings {
     authentication_mode: KubernetesAuthenticationMode,
     bootstrap_token: KubernetesBootstrapToken,
     standalone_mode: bool,
-    eviction_hard: HashMap<KubernetesEvictionHardKey, KubernetesThresholdValue>,
+    eviction_hard: HashMap<KubernetesEvictionKey, KubernetesThresholdValue>,
+    eviction_soft: HashMap<KubernetesEvictionKey, KubernetesThresholdValue>,
+    eviction_soft_grace_period: HashMap<KubernetesEvictionKey, KubernetesDurationValue>,
+    eviction_max_grace_period: i32,
```

Comment:
Do we need to constrain this value in anyway? Do we care about passing nonsensical values like `-100` to kubelet? We can use a modeled type to validate now that it supports scalar values.

Reply:
I think the only constraint is that it is a non-negative value.

Do you have an example of how to set that constraint?

Reply:
There is no existing setting that uses this yet but it is supported:
https://github.com/bottlerocket-os/bottlerocket/tree/develop/sources/models/scalar-derive#validating-a-number

Reply:
Maybe we can use regex to set the constraint? when I implemented eviction hard values, I used regex to limi the input value like this https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/models/src/modeled_types/kubernetes.rs#L538-L552

Reply:
We want to avoid using strings for value types that are supposed to be integers. Users have already complained about this inconsistency: https://github.com/bottlerocket-os/bottlerocket/issues/2883

Reply:
There are quite a few `i32` values in the kubernetes settings. I think nearly all of them have at least the constraint of needing to be non-negative. If we're going to enforce these constraints, we should probably do it for all of them.

Do we want to do that here, or would that be better kept it's only PR?

Reply:
Let's do it for this now. Need to start somewhere. The other ones were added before `scalar-derive` was a thing. That's on my plate to resolve.

Reply:
Error output isn't super great, but it does tell what the issue is:

```sh
[ssm-user@control]$ apiclient set --json '{"kubernetes": {"eviction-max-grace-period": -5}}'
Unable to deserialize input JSON into model: Unable to deserialize into PositiveInteger: number must be great than or equal to 0
[ssm-user@control]$ apiclient set --json '{"kubernetes": {"eviction-max-grace-period": 5}}'
[ssm-user@control]$ apiclient set --json '{"kubernetes": {"eviction-max-grace-period": 2147483647}}'
[ssm-user@control]$ apiclient set --json '{"kubernetes": {"eviction-max-grace-period": 2147483648}}'
Unable to deserialize input JSON into model: invalid value: integer `2147483648`, expected i32
```